mathex/mathex.go

```diff
@@ -0,0 +1,90 @@
+package mathex
+
+import (
+	"math"
+
+	"github.com/EliCDavis/vector"
+	"golang.org/x/exp/constraints"
+)
+
+type Number interface {
+	constraints.Integer | constraints.Float | vector.Number
+}
+
+func NearZero[T Number](v T) bool {
+	const s = 1e-8
+	return Abs(float64(v)) < s
+}
+
+func Npot[T Number](v T) T {
+	r := 1
+	for T(r) < v {
+		r <<= 1
+	}
+	return T(r)
+}
+
+// FloatEquals - Check whether two given floats are almost equal
+//func FloatEquals[FT FloatT](x, y FT) bool {
+//	return (mathex.Abs(x-y) <= 0.000001*math.Max(1.0, math.Max(mathex.Abs(x), mathex.Abs(y))))
+//}
+
+// Lerp - Calculate linear interpolation between two floats
+func Lerp[T Number](start, end T, amount float64) T {
+	return T(float64(start) + amount*float64(end-start))
+}
```

Comment:
Nit: I think I normally see the variable name as `time`.

Also, `time` should be before `start` and `end` to match the ordering of other function arguments like `Normalize`, `Wrap`, and `Clamp`.

```suggestion
func Lerp[T Number](time float64, start, end T) T {
	return T(float64(start) + time*float64(end-start))
}
```